lib/ansible/module_utils/facts.py

```diff
@@ -3169,8 +3169,44 @@ def populate(self):
         return self.facts
 
     def get_virtual_facts(self):
-        self.facts['virtualization_type'] = ''
-        self.facts['virtualization_role'] = ''
+        sysctl_path = self.module.get_bin_path('sysctl', required=True)
```

Comment:
This fails the whole fact gathering if missing, i would just add an `if sysctl_path:` instead